DFC.Composite.Shell/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -27,64 +39,263 @@ public static class ServiceCollectionExtensions
+                configuration);
+
+            services
+                .AddPolicies(
+                    policyRegistry,
+                    $"{nameof(AuthClientOptions)}_{nameof(PolicyOptions.HttpRetry)}",
+                    $"{nameof(AuthClientOptions)}_{nameof(PolicyOptions.HttpCircuitBreaker)}",
+                    policyOptions);
+
+            AddHttpClientWithPolicies<IApplicationHealthService, ApplicationHealthService, HealthClientOptions>(
+                services,
+                policyRegistry,
+                $"{nameof(HealthClientOptions)}_{nameof(PolicyOptions.HttpRetry)}",
+                $"{nameof(HealthClientOptions)}_{nameof(PolicyOptions.HttpCircuitBreaker)}",
+                nameof(HealthClientOptions),
+                policyOptions,
+                configuration);
+
+            AddHttpClientWithPolicies<IApplicationSitemapService, ApplicationSitemapService, SitemapClientOptions>(
+                services,
+                policyRegistry,
+                $"{nameof(SitemapClientOptions)}_{nameof(PolicyOptions.HttpRetry)}",
+                $"{nameof(SitemapClientOptions)}_{nameof(PolicyOptions.HttpCircuitBreaker)}",
+                nameof(SitemapClientOptions),
+                policyOptions,
+                configuration);
+
+            AddHttpClientWithPolicies<IApplicationRobotService, ApplicationRobotService, RobotClientOptions>(
+                services,
+                policyRegistry,
+                $"{nameof(RobotClientOptions)}_{nameof(PolicyOptions.HttpRetry)}",
+                $"{nameof(RobotClientOptions)}_{nameof(PolicyOptions.HttpCircuitBreaker)}",
+                nameof(RobotClientOptions),
+                policyOptions,
+                configuration);
+
+            AddHttpClientWithPolicies<IBannerService, BannerService, BannerClientOptions>(
+               services,
+               policyRegistry,
+               $"{nameof(BannerClientOptions)}_{nameof(PolicyOptions.HttpRetry)}",
+               $"{nameof(BannerClientOptions)}_{nameof(PolicyOptions.HttpCircuitBreaker)}",
+               nameof(BannerClientOptions),
+               policyOptions,
+               configuration);
+        }
+
+        private static bool AppRegistryRequestRegistered(this IServiceCollection services)
+        {
+            return services.Any(service => service.ServiceType == typeof(IAppRegistryService));
+        }
+
+        private static void AddApplicationClientHttp(
+            this IServiceCollection services,
+            IConfiguration configuration,
+            PolicyOptions policyOptions,
+            IPolicyRegistry<string> policyRegistry)
+        {
+            var registeredUrls = GetRegisteredUrls(services);
+
+            if (registeredUrls == null)
+            {
+                registeredUrls = new RegisteredUrls(default);
+            }
+
+            services.AddSingleton<IRegisteredUrls>(registeredUrls);
+
+            foreach (var registeredUrl in registeredUrls.GetAll())
```

Comment:
You're already looping through these to add the clients, just need to make them named clients.

Reply:
Made simpler now